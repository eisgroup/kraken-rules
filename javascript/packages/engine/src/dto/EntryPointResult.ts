/*
 *  Copyright 2018 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { FieldEvaluationResult } from "./FieldEvaluationResult";
import { RuleEvaluationResults } from "./RuleEvaluationResults";
import RuleEvaluationResult = RuleEvaluationResults.RuleEvaluationResult;

import { Reducer } from "declarative-js";
import flat = Reducer.flat;
import { ConditionEvaluationResult } from "./ConditionEvaluationResult";

/**
 * Contains evaluation results from specific entry point
 */
export class EntryPointResult {
    private readonly results: Record<string, FieldEvaluationResult>;
    readonly evaluationTimestamp: Date;
    constructor(results: Record<string, FieldEvaluationResult>) {
        this.results = results;
        this.getApplicableResults.bind(this);
        this.evaluationTimestamp = new Date();
    }
    /**
     * Flatten contents into single list of {@link RuleEvaluationResult}s
     */
    getAllRuleResults(): RuleEvaluationResult[] {
        return Object.values(this.results)
            .map(x => x.ruleResults)
            .reduce(flat, []);
    }
    getFieldResults(): Record<string, FieldEvaluationResult> {
        return this.results;
    }

    getApplicableResults(): RuleEvaluationResult[] {
        return this.getAllRuleResults()
            .filter(r => ConditionEvaluationResult.isApplicable(r.conditionEvaluationResult));
    }

    static empty(): EntryPointResult {
        return new EntryPointResult({});
    }
}
