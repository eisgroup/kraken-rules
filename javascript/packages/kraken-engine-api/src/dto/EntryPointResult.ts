/*
 *  Copyright 2022 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { FieldEvaluationResult } from './FieldEvaluationResult'
import { RuleEvaluationResults } from './RuleEvaluationResults'
import RuleEvaluationResult = RuleEvaluationResults.RuleEvaluationResult

/**
 * Contains evaluation results from specific entry point
 */
export interface EntryPointResult {
    readonly evaluationTimestamp: Date
    readonly ruleTimeZoneId: string

    /**
     * Returns entry point result mapped by field name.
     */
    getFieldResults(): Record<string, FieldEvaluationResult>

    /**
     * Flattens and returns field results into an array of {@link RuleEvaluationResult}
     */
    getAllRuleResults(): RuleEvaluationResult[]

    /**
     * Flattens and returns field results into an array of {@link RuleEvaluationResult}
     * filtered by rule condition evaluation status.
     */
    getApplicableResults(): RuleEvaluationResult[]
}
