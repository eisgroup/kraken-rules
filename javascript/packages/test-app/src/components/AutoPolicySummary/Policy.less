/*
 *  Copyright 2019 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

body {
    margin-top: 0px;
}

.input-width {
    width: 90% !important;
    margin-bottom: 15px !important;
    margin-top: 3px !important;
}

.entry-point-name-select {
    margin: 0 10px 0 0;
}

#buttonPanel {
    width: 100%;
    position: fixed;
    top: 0px;
    background: #1890ffe8;
    padding: 16px;
    right: 0px;
    z-index: 1;
    text-align: right;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.26);
}

#buttonPanel button {
    margin-right: 10px;
}

#buttonPanel div {
    margin-right: 10px;
}

.ContextDefinitionInfo {
    border-radius: 3px;
    padding: 11px;
    background: #90caff59;
    border: 1px solid #2b98fd5e;
    margin: 10px 0px;
}

.sub-entity-box {
    border: 3px solid #80808087;
    padding: 18px;
    margin: 18px;
    border-radius: 3px;
    box-shadow: 0px 0px 20px -3px rgba(0, 0, 0, 0.5);
    background: url('https://www.toptal.com/designers/subtlepatterns/patterns/dot-grid.png');
}
.sub-entity-box > h2 {
    color: rgb(45, 154, 255);
    text-align: center;
    text-transform: capitalize;
}
