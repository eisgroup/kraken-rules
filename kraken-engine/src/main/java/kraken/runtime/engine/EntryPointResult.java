package kraken.runtime.engine;

import kraken.annotations.API;
import kraken.runtime.engine.dto.FieldEvaluationResult;
import kraken.runtime.engine.dto.RuleEvaluationResult;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

/**
 * Contains evaluation results from specific entry point
 *
 * @author rimas
 * @since 1.0
 */
@API
public class EntryPointResult {

    /**
     * Field evaluation results, mapped to field id string
     */
    private Map<String, FieldEvaluationResult> fieldResults;

    private transient LocalDateTime evaluationTimeStamp;
    /**
     * Default constructor
     */
    public EntryPointResult() {
        fieldResults = new HashMap<>();
        evaluationTimeStamp = LocalDateTime.now();
    }

    public EntryPointResult(Map<String, FieldEvaluationResult> fieldResults, LocalDateTime evaluationTimeStamp) {
        this.fieldResults = new HashMap<>(fieldResults);
        this.evaluationTimeStamp = evaluationTimeStamp;
    }

    /**
     * Flatten contents into single list of {@link RuleEvaluationResult}s
     *
     * @return  list of all contained {@link RuleEvaluationResult}s
     */
    public List<RuleEvaluationResult> getAllRuleResults() {
        return fieldResults.entrySet().stream()
                .flatMap(e -> e.getValue().getRuleResults().stream())
                .collect(Collectors.toList());
    }

    public List<RuleEvaluationResult> getApplicableRuleResults() {
        return fieldResults.entrySet().stream()
                .flatMap(e -> e.getValue().getRuleResults().stream())
                .filter(r -> r.getConditionEvaluationResult().isApplicable())
                .collect(Collectors.toList());
    }

    public LocalDateTime getEvaluationTimeStamp() {
        return evaluationTimeStamp;
    }

    public Map<String, FieldEvaluationResult> getFieldResults() {
        return fieldResults;
    }

}
