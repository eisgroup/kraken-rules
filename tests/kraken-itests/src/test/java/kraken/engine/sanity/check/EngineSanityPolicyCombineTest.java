package kraken.engine.sanity.check;

import kraken.runtime.engine.EntryPointResult;
import kraken.testproduct.domain.Policy;
import kraken.utils.MockAutoPolicyBuilder;
import org.junit.BeforeClass;
import org.junit.Test;

import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static kraken.testing.matchers.KrakenMatchers.hasNoIgnoredRules;
import static kraken.testing.matchers.KrakenMatchers.hasNoValidationFailures;
import static kraken.testing.matchers.KrakenMatchers.hasRuleResults;
import static kraken.test.KrakenItestMatchers.matchesSnapshot;
import static org.hamcrest.MatcherAssert.assertThat;

public class EngineSanityPolicyCombineTest extends SanityEngineBaseTest {

    @BeforeClass
    public static void beforeAll() {
        start();
    }

    @Test
    public void shouldEvaluatePolicyCombinedEntryPoint() {
        Policy policy = new MockAutoPolicyBuilder().addValidAutoPolicy().build();
        EntryPointResult result = engine.evaluate(policy, "PolicyCombined");

        assertThat(result, hasRuleResults(32));
        assertThat(result, hasNoValidationFailures());
        assertThat(result, hasNoIgnoredRules());
        assertThat(result, matchesSnapshot());
    }
}
