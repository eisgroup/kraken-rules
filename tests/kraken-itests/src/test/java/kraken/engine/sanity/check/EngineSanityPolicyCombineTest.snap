kraken.engine.sanity.check.EngineSanityPolicyCombineTest.shouldEvaluatePolicyCombinedEntryPoint=[
  {
    "AddressInfo:-1:countryCd": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "countryCd",
        "fieldPath": "countryCd"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0156",
            "context": "AddressInfo",
            "targetPath": "countryCd",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        },
        {
          "ruleInfo": {
            "ruleName": "R0156",
            "context": "AddressInfo",
            "targetPath": "countryCd",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "AddressInfo:-1:doNotSolicit": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "doNotSolicit",
        "fieldPath": "doNotSolicit"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0155",
            "context": "AddressInfo",
            "targetPath": "doNotSolicit",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        },
        {
          "ruleInfo": {
            "ruleName": "R0155",
            "context": "AddressInfo",
            "targetPath": "doNotSolicit",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "AddressInfo:-1:postalCode": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "AddressInfo",
        "fieldName": "postalCode",
        "fieldPath": "postalCode"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0157",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        },
        {
          "ruleInfo": {
            "ruleName": "R0157",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "BillingAddress:-1:countryCd": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "BillingAddress",
        "fieldName": "countryCd",
        "fieldPath": "countryCd"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0156",
            "context": "AddressInfo",
            "targetPath": "countryCd",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "BillingAddress:-1:doNotSolicit": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "BillingAddress",
        "fieldName": "doNotSolicit",
        "fieldPath": "doNotSolicit"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0155",
            "context": "AddressInfo",
            "targetPath": "doNotSolicit",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: doNotSolicit \u003d true or doNotSolicit \u003d false",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "BillingAddress:-1:postalCode": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "BillingAddress",
        "fieldName": "postalCode",
        "fieldPath": "postalCode"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0157",
            "context": "AddressInfo",
            "targetPath": "postalCode",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Party:1:relationToPrimaryInsured": {
      "contextFieldInfo": {
        "contextId": "1",
        "contextName": "Party",
        "fieldName": "relationToPrimaryInsured",
        "fieldPath": "relationToPrimaryInsured"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0149",
            "context": "Party",
            "targetPath": "relationToPrimaryInsured",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Party:3:relationToPrimaryInsured": {
      "contextFieldInfo": {
        "contextId": "3",
        "contextName": "Party",
        "fieldName": "relationToPrimaryInsured",
        "fieldPath": "relationToPrimaryInsured"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0149",
            "context": "Party",
            "targetPath": "relationToPrimaryInsured",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "PartyRole:-1:role": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "PartyRole",
        "fieldName": "role",
        "fieldPath": "role"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0148",
            "context": "PartyRole",
            "targetPath": "role",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "PersonInfo:-1:firstName": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "PersonInfo",
        "fieldName": "firstName",
        "fieldPath": "firstName"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0150",
            "context": "PersonInfo",
            "targetPath": "firstName",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: firstName !\u003d lastName",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:contractTermTypeCd": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "contractTermTypeCd",
        "fieldPath": "termDetails.contractTermTypeCd"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0042",
            "context": "Policy",
            "targetPath": "contractTermTypeCd",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:createdBy": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "createdBy",
        "fieldPath": "accessTrackInfo.createdBy"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0032",
            "context": "Policy",
            "targetPath": "createdBy",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:createdFromPolicyRev": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "createdFromPolicyRev",
        "fieldPath": "createdFromPolicyRev"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0008",
            "context": "Policy",
            "targetPath": "createdFromPolicyRev",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:createdOn": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "createdOn",
        "fieldPath": "accessTrackInfo.createdOn"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0028",
            "context": "Policy",
            "targetPath": "createdOn",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:effectiveDate": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "effectiveDate",
        "fieldPath": "termDetails.termEffectiveDate"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0044",
            "context": "Policy",
            "targetPath": "effectiveDate",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:expirationDate": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "expirationDate",
        "fieldPath": "termDetails.termExpirationDate"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0047",
            "context": "Policy",
            "targetPath": "expirationDate",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:policyNumber": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "policyNumber",
        "fieldPath": "policyNumber"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0001A",
            "context": "Policy",
            "targetPath": "policyNumber",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "Field is required",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:state": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "state",
        "fieldPath": "state"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0004",
            "context": "Policy",
            "targetPath": "state",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:termCd": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "termCd",
        "fieldPath": "termDetails.termCd"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0054",
            "context": "Policy",
            "targetPath": "termCd",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:termNo": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "termNo",
        "fieldPath": "termDetails.termNo"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0051",
            "context": "Policy",
            "targetPath": "termNo",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: termNo \u003e\u003d 0 and termNo \u003c 100",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        },
        {
          "ruleInfo": {
            "ruleName": "R0050",
            "context": "Policy",
            "targetPath": "termNo",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:txCreateDate": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "txCreateDate",
        "fieldPath": "transactionDetails.txCreateDate"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0020",
            "context": "Policy",
            "targetPath": "txCreateDate",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:txEffectiveDate": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "txEffectiveDate",
        "fieldPath": "transactionDetails.txEffectiveDate"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0016",
            "context": "Policy",
            "targetPath": "txEffectiveDate",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:txType": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "txType",
        "fieldPath": "transactionDetails.txType"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0011",
            "context": "Policy",
            "targetPath": "txType",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:updatedBy": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "updatedBy",
        "fieldPath": "accessTrackInfo.updatedBy"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0039",
            "context": "Policy",
            "targetPath": "updatedBy",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Policy:-1:updatedOn": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Policy",
        "fieldName": "updatedOn",
        "fieldPath": "accessTrackInfo.updatedOn"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0035",
            "context": "Policy",
            "targetPath": "updatedOn",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Vehicle:-1:included": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Vehicle",
        "fieldName": "included",
        "fieldPath": "included"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0073",
            "context": "Vehicle",
            "targetPath": "included",
            "payloadType": "ASSERTION"
          },
          "payloadResult": {
            "success": true,
            "messageCode": "code",
            "message": "Value didn\u0027t match asserton: included \u003d true or included \u003d false",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Vehicle:-1:model": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Vehicle",
        "fieldName": "model",
        "fieldPath": "model"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0074",
            "context": "Vehicle",
            "targetPath": "model",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    },
    "Vehicle:-1:purchasedDate": {
      "contextFieldInfo": {
        "contextId": "-1",
        "contextName": "Vehicle",
        "fieldName": "purchasedDate",
        "fieldPath": "purchasedDate"
      },
      "ruleResults": [
        {
          "ruleInfo": {
            "ruleName": "R0088",
            "context": "Vehicle",
            "targetPath": "purchasedDate",
            "payloadType": "USAGE"
          },
          "payloadResult": {
            "usageType": "mandatory",
            "success": true,
            "messageCode": "code",
            "message": "The field is mandatory",
            "templateVariables": [],
            "validationSeverity": "critical"
          },
          "conditionEvaluationResult": {
            "conditionEvaluation": "APPLICABLE"
          },
          "overrideInfo": {
            "isOverridable": false
          }
        }
      ]
    }
  }
]